torch/fx/experimental/param_fetch.py

```diff
@@ -0,0 +1,42 @@
+from torch.fx.graph_module import GraphModule
+from typing import Any, Dict, List, Tuple
+import torch.nn as nn
+
+
+"""This dict maps the nn.Module class name to the attribute name list that we want to fetch for lowering.
+The first integer in the tuple is the version number of the nn.Module class when we create the parameter list.
+If there's a version mismatch then it means the parameter names in the book might be mismatched with nn.Module.
+"""
+module_fetch_book: Dict[str, Tuple[int, List[str]]] = {
```

Comment:
You should key on the `Type`s themselves, not their string `__name__`:

```
from typing import Type

<...>

module_fetch_book: Dict[Type, Tuple[int, List[str]]] = {
    torch.nn.Linear: (1, ["weight", "bias"]),
    torch.nn.Conv2d: (1, ["weight", "bias", "kernel_size", "stride", "padding", "dilation", "groups", "padding_mode"]),
    torch.nn.BatchNorm2d: (2, ["weight", "bias", "running_mean", "running_var"]),
}
```

Using `__name__` isn't safe, because it's only the last atom of the qualified name. Alternatively we could match on `torch.typename(m)`, but it's probably better to not project into string space at all and just match directly on types